stdlib/typing.pyi

```diff
@@ -597,12 +597,23 @@ class Match(Generic[AnyStr]):
     if sys.version_info >= (3, 9):
         def __class_getitem__(cls, item: Any) -> GenericAlias: ...
 
+# Not actually defined in typing (or re!) - is _sre.SRE_Scanner at runtime
+# Also not documented anywhere
+@_final
+class _SRE_Scanner(Generic[AnyStr]):
+    pattern: Pattern[AnyStr]
+    # Cannot be directly instantiated
+    def __new__(cls, *args: Any, **kwargs: Any) -> NoReturn: ...  # type: ignore
```

Comment:
Why does this need `# type: ignore`?

Reply:
Because `__new__` has to return an instance of `cls`, according to normal rules.